src/validation.cpp

```diff
@@ -2585,8 +2602,8 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
         {
             if (tx.vout[j].scriptPubKey == Params().GetConsensus().burnAddress)
             {
-                pburnHistoryDB->WriteAccountHistory({Params().GetConsensus().burnAddress, static_cast<uint32_t>(pindex->nHeight), i},
-                                            {block.vtx[i]->GetHash(), static_cast<uint8_t>(CustomTxType::None), {{DCT_ID{0}, tx.vout[j].nValue}}});
+                pburnHistoryDB->WriteAccountHistory({tx.vout[j].scriptPubKey, static_cast<uint32_t>(pindex->nHeight), i},
+                                                    {block.vtx[i]->GetHash(), static_cast<uint8_t>(CustomTxType::None), {{DCT_ID{0}, tx.vout[j].nValue}}});
```

Comment:
Will this cause the issue "`AccountHistory.blockHash` returned from `getaccounthistory` becomes undefined" ?
Related jellyfish test: https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/__tests__/category/account/getAccountHistory.test.ts#L46

Reply:
No, it shouldn't